### PR TITLE
[FIX] project: fix project privacy_visibility tooltip

### LINKED ISSUE #92734: [14.0] Project: wrong description compared to features
Screenshot from 2022-06-02 14-29-40

Current behavior: **Invited portal and all internal users:** `employees may see everything. Portal users may see project and tasks followed by them or by someone of their company.` In fact, those who use the portal with their company but are not followed will not be able to see this project and task.

Support ticket number submitted via odoo.com/help (optional):

### PR DESCRIPTION
**Pull request by MissingNoShiny**
The project privacy_visibility tooltip doesn't reflect the actual behavior for the "Invited portal users and all internal users" selection.

This PR rephrases the tooltip of that field in order to correctly match its behavior.
It also slightly changes the name of the options to match the new tooltip.

Fixes #92734

Task-2871771

### CODE REVIEW

**Comment by MissingNoShiny** [comment]
Hello @tranngocson1996, I was busy with other stuff this week, but this PR (and another one for helpdesk, since I found out there was the same tooltip issue with helpdesk teams) is now ready for review, so it should hopefully not take too long until it gets merged. :slightly_smiling_face:

**Comment by tranngocson1996** [comment]
> Hello @tranngocson1996, I was busy with other stuff this week, but this PR (and another one for helpdesk, since I found out there was the same tooltip issue with helpdesk teams) is now ready for review, so it should hopefully not take too long until it gets merged. :slightly_smiling_face: 

@MissingNoShiny thank you bro

File: addons/project/i18n/project.pot [hunk 1]
```diff
@@ -6,8 +6,8 @@ msgid ""
 msgstr ""
 "Project-Id-Version: Odoo Server 13.0\n"
 "Report-Msgid-Bugs-To: \n"
-"POT-Creation-Date: 2020-04-27 09:21+0000\n"
-"PO-Revision-Date: 2020-04-27 09:21+0000\n"
+"POT-Creation-Date: 2022-06-03 14:56+0000\n"
+"PO-Revision-Date: 2022-06-03 14:56+0000\n"
```
Comment (IT-Ideas) [inline addons/project/i18n/project.pot:10]: I am unsure we really need to update those fields... I have just asked @mart-e 
Let's wait for his answer 🙂

File: addons/project/i18n/project.pot [hunk 2]
```diff
@@ -90,7 +90,11 @@ msgid ""
 "    <table border=\"0\" cellpadding=\"0\" cellspacing=\"0\" width=\"590\" style=\"width:100%; margin:0px auto;\">\n"
 "    <tbody>\n"
 "        <tr><td valign=\"top\" style=\"font-size: 13px;\">\n"
-"            Hello ${partner.name},<br/>\n"
+"            % if partner.name:\n"
+"                Hello ${partner.name},<br/><br/>\n"
+"            % else:\n"
+"                Hello,<br/><br/>\n"
+"            % endif\n"
```
Comment (IT-Ideas) [inline addons/project/i18n/project.pot:97]: Shouldn't we add this in the commit description?

File: addons/project/i18n/project.pot [hunk 3]
```diff
@@ -1931,6 +1930,22 @@ msgid ""
...
+"A user with the project > administrator access right level can still access this project and its tasks, even if they are not explicitly part of the followers.\n"
+"\n"
+"- All internal users: all internal users can access the project and all of its tasks without distinction.\n"
+"\n"
+"- Invited portal users and all internal users: all internal users can access the project and all of its tasks without distinction.\n"
+"Portal users can only access the projects and the tasks they are following. This access can be modified on each task individually by adding or removing the portal user as follower.\n"
+"\n"
+"In any case, an internal user with no project access rights can still access a task, provided that he is given the corresponding URL (and that he is part of the followers if the project is private)."
+msgstr ""
+
```
Comment (IT-Ideas) [inline addons/project/i18n/project.pot:1948]: Ideally, in order to keep the diff minimal, I would not change the place of this traduction in the file.

File: addons/project/i18n/project.pot [hunk 4]
```diff
@@ -1401,7 +1395,12 @@ msgstr ""
 
 #. module: project
 #: model:ir.model.fields.selection,name:project.selection__project_project__privacy_visibility__followers
-msgid "Invited employees"
+msgid "Invited internal users"
+msgstr ""
+
+#. module: project
+#: model:ir.model.fields.selection,name:project.selection__project_project__privacy_visibility__portal
+msgid "Invited portal users and all internal users"
```
Comment (IT-Ideas) [inline addons/project/i18n/project.pot:1403]: Ideally, in order to keep the diff minimal, I would not change the place of this traduction in the file.

**Review by IT-Ideas — COMMENTED** [review]
Thanks 🙂

File: addons/project/i18n/project.pot (hunk 1 above)
Comment (IT-Ideas) [reply addons/project/i18n/project.pot:10]: I have the confirmation that this diff can be removed 🙂

**Review by IT-Ideas — APPROVED** [review]
Thanks 🙂

**Comment by IT-Ideas** [comment]
@robodoo r+

**Comment by IT-Ideas** [comment]
@robodoo r+

File: addons/project/i18n/project.pot [hunk 5]
```diff
@@ -888,11 +888,17 @@ msgstr ""
...
 msgid ""
-"Defines the visibility of the tasks of the project:\n"
-"- Invited employees: employees may only see the followed project and tasks.\n"
-"- All employees: employees may see all project and tasks.\n"
-"- Portal users and all employees: employees may see everything.   Portal users may see project and tasks followed by.\n"
-"   them or by someone of their company."
+"People to whom this project and its tasks will be visible.\n"
+"\n"
+"- Invited internal users: when following a project, internal users will get access to all of its tasks without distinction. Otherwise, they will only get access to the specific tasks they are following.\n"
+" A user with the project > administrator access right level can still access this project and its tasks, even if they are not explicitly part of the followers.\n"
```
Comment (minhkuheo) [inline addons/project/i18n/project.pot:894]: Would you please tell me what this symbol ">" means in this context?

Comment (IT-Ideas) [reply addons/project/i18n/project.pot:894]: This means we talk about the `the administrator access right level` `of` `project`.

File: addons/project/models/project.py [hunk 6]
```diff
@@ -190,18 +190,22 @@ def _get_default_favorite_user_ids(self):
...
+            "- Invited internal users: when following a project, internal users will get access to all of its tasks without distinction. "
+            "Otherwise, they will only get access to the specific tasks they are following.\n "
+            "A user with the project > administrator access right level can still access this project and its tasks, even if they are not explicitly part of the followers.\n\n"
+            "- All internal users: all internal users can access the project and all of its tasks without distinction.\n\n"
+            "- Invited portal users and all internal users: all internal users can access the project and all of its tasks without distinction.\n"
+            "Portal users can only access the projects and the tasks they are following. "
+            "This access can be modified on each task individually by adding or removing the portal user as follower.\n\n"
+            "In any case, an internal user with no project access rights can still access a task, "
+            "provided that they are given the corresponding URL (and that they are part of the followers if the project is private).")
```
Comment (Thaotran0967) [inline addons/project/models/project.py:208]: an internal user with no project access rights can't access the task even though the URL is shared and they are part of the followers
@tranngocson1996

Comment (MissingNoShiny) [reply addons/project/models/project.py:208]: Hello, it seems like it doesn't work like that in version 14 indeed, but only in that version, curiously. We will make a fix, thanks for the follow-up.

Comment (sang250399) [inline addons/project/models/project.py:208]: i don't understand this line, can you help me explain? @MissingNoShiny

Comment (MissingNoShiny) [reply addons/project/models/project.py:208]: Hello, it means that if you give the url of the task to an internal user, they can view it regardless of their project access rights. If the project has the visibility "Invited internal users" however, they still need to be part of the followers of the task for this to apply.

Comment (sang250399) [reply addons/project/models/project.py:208]: > Hello, it means that if you give the url of the task to an internal user, they can view it regardless of their project access rights. If the project has the visibility "Invited internal users" however, they still need to be part of the followers of the task for this to apply.

oke, i understand,thanks !

Comment (MissingNoShiny) [reply addons/project/models/project.py:208]: You're welcome :slightly_smiling_face:

**Pull request closed without merging**